contracts/exchangeIssuance/ExchangeIssuance.sol

```diff
@@ -0,0 +1,384 @@
+     * @param _amountInput      Amount of the input token to spend
+     * @param _isInputETH       Set to true if the input token is Ether
+     * @param _inputToken       Address of input token. Ignored if _isInputETH is true
+     * @param _minSetReceive    Minimum amount of index to receive
+     */
+    function exchangeIssue(ISetToken _setToken, uint256 _amountInput, bool _isInputETH, IERC20 _inputToken, uint256 _minSetReceive) external payable nonReentrant {
+        if(_isInputETH) {
+            require(msg.value == _amountInput, "INCORRECT_INPUT_AMOUNT");
+            IWETH(WETH).deposit{value: msg.value}();    // ETH -> WETH
+        } else if(address(_inputToken) != WETH) {
+            _inputToken.transferFrom(msg.sender, address(this), _amountInput);
+            purchaseWETHExactTokens(address(_inputToken), _amountInput);    // _inputToken -> WETH
+        }
+
+        // get price of set token on uniswap
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        (uint256 tokenReserveA, uint256 tokenReserveB) = UniswapV2Library.getReserves(uniFactory, WETH, address(_setToken));
+        uint256 minSetTokenAmountOut = uniRouter.getAmountOut(wethBalance, tokenReserveA, tokenReserveB);
+
+        uint256 sumEth = 0;
+        ISetToken.Position[] memory positions = _setToken.getPositions();
+        uint256[] memory amountEthIn = new uint256[](positions.length);
+        for(uint256 i = 0; i < positions.length; i++) {
+            uint256 unit = uint256(positions[i].unit);       // amount of token per 1 index token
+            address token = positions[i].component;
+            uint256 amountOut = minSetTokenAmountOut * unit / 1 ether;
+            uint256 uniPrice = tokenAvailable(uniFactory, token) ? getBuyPrice(true, token, amountOut) : PreciseUnitMath.maxUint256();
+            uint256 sushiPrice = tokenAvailable(sushiFactory, token) ? getBuyPrice(false, token, amountOut) : PreciseUnitMath.maxUint256();
+            uint256 amountEth = Math.min(uniPrice, sushiPrice);
+            sumEth += amountEth;
+            amountEthIn[i] = amountEth;
+        }
+
+        uint256 maxIndexAmount = acquireComponents(positions, amountEthIn, wethBalance, sumEth);
+        require(maxIndexAmount > _minSetReceive, "INSUFFICIENT_OUTPUT_AMOUNT");
+        basicIssuanceModule.issue(_setToken, maxIndexAmount, msg.sender);
+    }
+
+    /**
+     * Get approx output amount when issuing using the
+     * exchangeIssue function.
+     *
+     * @param _setToken         Address of the set token being issued
+     * @param _amountInput      Amount of the input token to spend
+     * @param _isInputETH       Set to true if the input token is Ether
+     * @param _inputToken       Address of input token. Ignored if _isInputETH is true
+     */
+    function getExchangeIssue(ISetToken _setToken, uint256 _amountInput, bool _isInputETH, IERC20 _inputToken) external view returns (uint256) {
```

Comment:
Same comment re: return javadoc and order.

For function clarity, consider renaming to `getEstimatedIssueSetQuantity`